core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -800,27 +839,39 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
 }
 
   @Test
-  def testCreatePermissionNeededToReadFromNonExistentTopic() {
-    val newTopic = "newTopic"
+  def testCreatePermissionOnTopicToReadFromNonExistentTopic() {
+    testCreatePermissionNeededToReadFromNonExistentTopic("newTopic",
+      Set(new Acl(userPrincipal, Allow, Acl.WildCardHost, Create)),
+      Topic)
+  }
+
+  @Test
+  def testCreatePermissionOnClusterToReadFromNonExistentTopic() {
+    testCreatePermissionNeededToReadFromNonExistentTopic("newTopic",
+      Set(new Acl(userPrincipal, Allow, Acl.WildCardHost, Create)), 
+      Cluster)
+  }
+
+  private def testCreatePermissionNeededToReadFromNonExistentTopic(newTopic: String, acls: Set[Acl], resType: ResourceType) {
     val topicPartition = new TopicPartition(newTopic, 0)
     val newTopicResource = new Resource(Topic, newTopic)
     addAndVerifyAcls(Set(new Acl(userPrincipal, Allow, Acl.WildCardHost, Read)), newTopicResource)
     addAndVerifyAcls(groupReadAcl(groupResource), groupResource)
-    addAndVerifyAcls(clusterAcl(Resource.ClusterResource), Resource.ClusterResource)
+    this.consumers.head.assign(List(topicPartition).asJava)
     try {
-      this.consumers.head.assign(List(topicPartition).asJava)
-      consumeRecords(this.consumers.head)
-      Assert.fail("should have thrown exception")
+      this.consumers.head.poll(Duration.ofMillis(50L));
```

Comment:
Why did we replace `consumeRecords` with `poll`?

Reply:
We modified the method to avoid producing any records - to avoid adding write permission which made the test more complicated than it needed to be.

So we need to pass `numRecords=0` and we had no control on the timeout, unless we made that another optional argument.
without actually consuming records it seems to me that just calling poll is simpler than consumeRecords

Reply:
We want to be generous with the timeout then. Any reason why it's so low?

Reply:
the default timeout was fine for the first call (when we expect it always to throw)

it's the second one that caused issues - see https://issues.apache.org/jira/browse/KAFKA-6994 - but that IMHO is unrelated to the changes in this PR.

if the second timeout is low - even without adding the required acl the test passed - that's the bug 6994.

if the second timeout is large the test takes longer, that's why we kept relatively short

Reply:
Did you see my comment in KAFKA-6994?

Reply:
yep - replied there - it's the same with the old `poll`

Reply:
Even then, `poll` provides little in the way of guarantees. I suggest using `TestUtils.retry` with a generous timeout and a short timeout for `poll`. This also ensures we finish as quickly as possible.

Reply:
good point - we can use `waitUntilTrue` checking for the topic being created as a consequence of `poll`